### PR TITLE
Geo/thermal tests documentation

### LINKED ISSUE #11872: [GeoMechanicsApplication] Thermal tests documentation
What is added in this PR:

- README.md files are added for each type of thermal tests
- Added test_transient_heat_flux_2D3N test as it was missing
- The thermal tests are now categorized in folders based on their type of test
- "Unsteady" tests name changed to "Transient"
- Boundary condition figures are changed according to team suggestion
- Result animations are for the transient test cases with a fixed legend
- Graph for results for steady state tests are added
- "init" condition name is changed to "initial" for all tests

### PR DESCRIPTION
**Pull request by indigocoral**
**📝 Description**
Added README.md files for the thermal tests and made some changes to the tests

**🆕 Changelog**

- README.md files are added for each type of thermal tests
- Added test_transient_heat_flux_2D3N test as it was missing
- The thermal tests are now categorized in folders based on their type of test
- Boundary condition figures are changed according to team suggestion
- Result animations are for the transient test cases with a fixed legend
- Graph for results for steady state tests are added
- "init" condition name is changed to "initial" for all tests

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_fixed_temperature/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Test Cases for Thermal fixed temperature
+
+**Author:** [Mohamed Nabi](https://github.com/mnabideltares)
+
+**Source files:** [Thermal fixed temperature](https://github.com/KratosMultiphysics/Kratos/tree/master/applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_fixed_temperature)
+
+
+## Case Specification
+In this thermal test case, a 1 x 1 m soil is considered, with the left side set at 0 degrees and the right side set at 25 degrees. The simulation spans 100 days to allow for a transition from an exponential to a linear temperature profile between the two sides. This test is conducted for various configurations, including 2D3N, 2D4N, 2D6N, 2D8N, 2D9N, 2D10N, 2D15N, 3D4N, and 3D10N. The temperature distribution within the box is then evaluated with its own result.
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_fixed_temperature/README.md]: The initial profile is not an exponential one, it is a step function ( starts from 0 everywhere, then a sudden jump to 25 at the right boundary ).

File: applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Test Cases for Thermal heat flux
+
+**Author:** [Mohamed Nabi](https://github.com/mnabideltares)
+
+**Source files:** [Thermal heat flux](https://github.com/KratosMultiphysics/Kratos/tree/master/applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux)
+
+
+## Case Specification
+In this thermal heat flux test case, a 1 x 1 m soil box is considered, with a 100 J/s.m2 heat flux applied as a condition to the bottom side and the top temperature set to 0 degrees. The simulation spans 100 days to allow for a transition from an exponential to a linear temperature profile between the two sides. This test is conducted for various configurations, including 2D3N, 2D4N, 2D6N, 2D8N, 2D9N, 2D10N, 2D15N, 3D4N, and 3D10N. The temperature distribution within the box is then evaluated with its own result.
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md]: I wonder if the J/s.m2 displays as intended. Both the time and length squared are below the division sign. the 2 is a power and hopefully a superscript.

**Review by WPK4FEM — COMMENTED** [review]
Nice catch, the lingering MainKratos.py
Thank you for bringing more consistency and the nice documentation.

**Review by WPK4FEM — DISMISSED** [review]
Nicer formatting!

File: applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_transient_thermal_heat_flux_2D9N_result.gif
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_transient_thermal_heat_flux_2D9N_result.gif:1]: Is it possible to slow down the picture? Currently, it is too fast to follow.

File: applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_transient_thermal_fixed_temperature_2D9N_result.gif
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_transient_thermal_fixed_temperature_2D9N_result.gif:1]: It is a question to Mohamed, shall we explain why temperature is below zero at a certain time?

File: applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_thermal_fixed_temperature_2D9N_result.png
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_thermal_fixed_temperature_2D9N_result.png:1]: I would change the scale to have from 0 to 25. Currently, the picture confuses at the first glance. It takes time to see that the temperature value at the right is 25 C.

File: applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Test Cases for Thermal heat flux
+
+**Author:** [Mohamed Nabi](https://github.com/mnabideltares)
+
+**Source files:** [Thermal heat flux](https://github.com/KratosMultiphysics/Kratos/tree/master/applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux)
+
+## Case Specification
+In this thermal heat flux test case, a 1 x 1 m soil box is considered, with a 100 J.s<sup>-1</sup>.m<sup>-2</sup> heat flux applied as a condition to the bottom side and the top temperature set to 0 degrees. The simulation spans 100 days to allow for a transition from an exponential to a linear temperature profile between the two sides. This test is conducted for various configurations, including 2D3N, 2D4N, 2D6N, 2D8N, 2D9N, 2D10N, 2D15N, 3D4N, and 3D10N. The temperature distribution within the box is then evaluated with its own result.
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md]: The picture with temperature above shows data for 2.17e7 seconds that is 250 days.

File: applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_fixed_temperature/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+## Case Specification
+In this thermal test case, a 1 x 1 m soil is considered, with everywhere set to 0 degrees and then a sudden jump to 25 degrees at the right boundary. The simulation spans 100 days to allow for a transition from an exponential to a linear temperature profile between the two sides. This test is conducted for various configurations, including 2D3N, 2D4N, 2D6N, 2D8N, 2D9N, 2D10N, 2D15N, 3D4N, and 3D10N. The temperature distribution within the box is then evaluated with its own result.
+The boundary conditions are shown below:
+
+<img src="../documentation_data/test_fixed_temperature_boundary_conditions.png" alt="Visualization of the Boundary conditions" title="Visualization of the Boundary conditions" width="600">
+
+## Results
+
+The animated figure below illustrates the temperature contours resulting from the simulation (as an example the 2D9N test is shown below).
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_fixed_temperature/README.md]: I think it shall be 'picture' instead of 'animated figure'. Correct?

File: applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+The boundary conditions are shown below:
+
+<img src="../documentation_data/test_heat_flux_boundary_conditions.png" alt="Visualization of the Boundary conditions" title="Visualization of the Boundary conditions" width="600">
+
+## Results
+
+The animated figure below illustrates the temperature contours resulting from the simulation (as an example the 2D9N test is shown below).
```
Comment (markelov208) [inline applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md]: As above I think it shall be 'picture' instead of 'animated figure'. Correct?

**Review by markelov208 — COMMENTED** [review]
Hi Marjan, a lot of work is done to have a good description. Just a few minor comments.

File: applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md (hunk 3 above)
Comment (indigocoral) [reply applications/GeoMechanicsApplication/tests/test_thermal_element/test_thermal_heat_flux/README.md]: Thank you, Gennady! I missed this one.

File: applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_transient_thermal_fixed_temperature_2D9N_result.gif
Comment (indigocoral) [reply applications/GeoMechanicsApplication/tests/test_thermal_element/documentation_data/test_transient_thermal_fixed_temperature_2D9N_result.gif:1]: Together with Wijtze Pieter, we added a short explanation in the README files.

**Review by markelov208 — APPROVED** [review]
Marjan, congrats with the first work in the new year!
